File: src/cli/commands/mcp.ts
import { Command } from 'commander'

interface GlobalOpts {
  db?: string
}

function getOpts(cmd: Command): GlobalOpts {
  let root = cmd
  while (root.parent) root = root.parent
  return root.opts()
}

export function registerMcpCommands(program: Command): void {
  const mcp = program.command('mcp').description('MCP server for AI agents')

  mcp
    .command('serve')
    .description('Start MCP server (stdio transport)')
    .action(async (_opts: Record<string, unknown>, cmd: Command) => {
      const globalOpts = getOpts(cmd)
      const { startMcpServer } = await import('../../mcp/server.js')
      await startMcpServer(globalOpts.db)
    })
}

File: internal/cli/mcp.go
package cli

import (
	crmmcp "github.com/jdanielnd/crm-cli/internal/mcp"
	"github.com/mark3labs/mcp-go/server"
	"github.com/spf13/cobra"
)

func registerMCPCommands(rootCmd *cobra.Command) {
	mcpCmd := &cobra.Command{
		Use:   "mcp",
		Short: "MCP server for AI agent integration",
	}

	mcpCmd.AddCommand(&cobra.Command{
		Use:   "serve",
		Short: "Start the MCP server (stdio transport)",
		RunE: func(cmd *cobra.Command, args []string) error {
			db, err := openDB()
			if err != nil {
				return err
			}
			defer db.Close()

			s := crmmcp.NewServer(db, Version)
			return server.ServeStdio(s)
		},
	})

	rootCmd.AddCommand(mcpCmd)
}
